Added a '--cache-only' feature for installation.

DigitalInBlue opened the pull request:
I added a "--cache-only" feature for installation.  This is needed because I have a use case where my development network is not connected to the Internet.  The machines that ARE connected to the Internet are not where I need to actually build.  I want to create a cache of all required packages, move it to the development network, and use the cached packages to build at that point.

Commit:
initial remove-in-parallel

doesn't actually do parallel remove yet

Commit:
remove_all parallelized, and fix the issues with symlink

Commit:
delete the random number generator

Commit:
actually get the code compiling

Commit:
make this compile on macos

Commit:
make it compile on macos under g++6

Commit:
fix some comments from code reviewers

Commit:
fix more comments

Commit:
add tests!

Also, fix all the bugs I found when I wrote the tests!

Commit:
forgot to test on macos >.<

Commit:
modify files test to include symlinks

Commit:
wheeeee more fixes

strega-nil committed:
tests.files.cpp:create_directory_tree -- change magic numbers to names

Commit:
clang-format all the things

Commit:
Merge branch 'trunk' into parallel-file-ops

Commit:
switch to new test framework

strega-nil committed:
allow tests to run on older standard libraries

strega-nil committed:
fix some of the awful mix of macros

DigitalInBlue committed:
Added a '--cache-only' feature for installation.

DigitalInBlue committed:
Added a '--cache-only' feature for installation.

cbezault commented:
While I agree with some of the whitespace changes that you made please only make them as a drive-by on lines or sections that you actually have to touch for your feature, it makes it easier to look at the diff.

DigitalInBlue commented:
Agree on the whitespace changes.  That was visual studio applying your .clang-format, not myself making manual changes.

cbezault commented:
Hmmm, I also don't see why you can't just use the `export` command to get NuGet packages to use on your isolated machines.

DigitalInBlue commented:
I need support for Windows and Linux, multiple compilers.  I also don't have a computer that is very good attached to the Internet.  Building on it is extremely painful.  We maintain a vcpkg cache on our disconnected network to build from on actual development machines.

cbezault commented:
Okay, it wasn't immediately clear from the dirty diff but now I see that you don't actually build, you really just get all the source. In that case I like this but maybe I'll ask the other people on the team if they can think of a more intuitive name than `--cache-only`.

DigitalInBlue commented:
Correct.  There is no building.  This feature just downloads all of the necessary packages into the ```Downloads``` folder.  We then take this folder to our development network and continue on.  

No pride in ownership.  This has been a great feature for our team and specific use case.

cbezault commented:
And I think it's a useful feature that I'd like to see mainlined. Thanks for the PR!

vicroms commented on toolsrc/include/vcpkg/build.h:
```diff
@@ -82,6 +82,12 @@ namespace vcpkg::Build
         YES
     };
 
+	enum class CacheOnly
```
Use spaces for indentation instead of tabs.
This should be consistent with the rest of the project.

DigitalInBlue commented:
I will update the formatting with your ```.clang-format```.  Do you want me to change the name of the flag as well?

vicroms commented:
I would rather have a new command `vcpkg download <package>` instead of just adding more options on top of `vcpkg install`.

DigitalInBlue commented:
```vcpkg download``` sounds reasonable to me.  I'll work that and get the change pushed back up.  Do you want a new pull request or just an update to this one?

cbezault commented:
Just update this one.

Commit:
[date] fix issue 6530 (#7399)

NancyLi1013 committed:
[xtensor-io] Add new port. (#7398)

LilyWangL committed:
[libmariadb] Fix install path (#7396)

tntxtnt committed:
[inja] Use inja CMakeLists.txt (#7402)

* Provide CMake configs

* Fix mismatching version in CMakeLists.txt (2.0.0 -> 2.1.0)

* Update license (use LICENSE as copyright instead of using single header file as copyright)

Cheney-W committed:
[Openxr-loader] Fix fatal error C1189 (#7376)

LilyWangL committed:
[tesseract[training_tools]] Fix build error (#7360)

contre committed:
[pcl] Fix cuda building compatability issues with cuda 10.1 (#7388)

strega-nil committed:
Merge pull request #7228 from ubsan/parallel-file-ops

Parallel file operations

strega-nil committed:
Add third party notices -- copied from chakracore (#7403)

Add third party notices -- copied from chakracore

strega-nil committed:
Modify CMakeLists to split up vcpkglib (#7407)

Before this, we had vcpkg and vcpkg-test taking the vcpkglib source
files directory -- this commit makes vcpkglib an object library.

xorz57 committed:
[EasyCL] New GitHub Repository (#7387)

* Add EasyCL GitHub Repository

* Update CONTROL

philjdf committed:
[Thrift] Make Thrift static again (#7302)

* Make thrift static

* Update CONTROL

* Thrift able to build statically and dynamically

* Clean up bin dir in static

* Changes as per discussion

* Remove comment

* Better comment

xorz57 committed:
[forest] Update to Version 12.0.3 (#7410)

* Update CONTROL

* Update to version 12.0.3

bobk2824 committed:
[ECSUtil] update library to v1.0.7.2 (#7427)

DigitalInBlue committed:
Added the Download command.

DigitalInBlue committed:
Merged upstream

DigitalInBlue committed:
Fixed merge errors.

DigitalInBlue commented:
This PR is failing and I've got a few things to clean up.  I am going to close this PR, clean up my fork, and initiate a new PR.  I will reference this PR in my new PR's comments.
